util/types/mytime.go

```diff
@@ -112,6 +112,58 @@ func newMysqlTime(year, month, day, hour, minute, second, microsecond int) mysql
 	}
 }
 
+func calcTimeFromSec(to *mysqlTime, seconds, microseconds int) {
+	to.hour = uint8(seconds / 3600)
+	seconds = seconds % 3600
+	to.minute = uint8(seconds / 60)
+	to.second = uint8(seconds % 60)
+	to.microsecond = uint32(microseconds)
+}
+
+// calcTimeDiff calculates difference between two datetime values as seconds + microseconds.
+// t1 and t2 should be TIME/DATE/DATETIME value.
+// sign can be +1 or -1, and t2 is preprocessed with sign first.
+func calcTimeDiff(t1, t2 TimeInternal, sign int) (seconds, microseconds int, neg bool) {
+	const secondsIn24Hour = 86400
+	days := calcDaynr(t1.Year(), t1.Month(), t1.Day())
+	days -= sign * calcDaynr(t2.Year(), t2.Month(), t2.Day())
```

Comment:
MYSQL_TIMESTAMP_TIME can use Go time method, this function just for DATETIME DATE TIME.